tests/artifactorydownload_test.go

```diff
@@ -200,4 +231,49 @@ func excludePatternsDownload(t *testing.T) {
 	if fileutils.IsPathExists(filepath.Join(workingDir, "b.in"), false) {
 		t.Error("File b.in should have been excluded")
 	}
+	if fileutils.IsPathExists(filepath.Join(workingDir, "c.tar.gz"), false) {
+		t.Error("File c.tar.gz should have been excluded")
+	}
+}
+
+func explodeArchiveDownload(t *testing.T) {
+	workingDir, err := ioutil.TempDir("", "downloadTests")
+	if err != nil {
+		t.Error(err)
+	}
+	defer os.RemoveAll(workingDir)
+	downloadPattern := RtTargetRepo + "*.tar.gz"
+	downloadTarget := workingDir + string(filepath.Separator)
+	downloadParams := services.DownloadParams{ArtifactoryCommonParams: &utils.ArtifactoryCommonParams{Pattern: downloadPattern, Recursive: true, Target: downloadTarget}, Flat: true, Explode: false}
+	_, _, err = testsDownloadService.DownloadFiles(downloadParams)
+	if err != nil {
+		t.Error(err)
+	}
+	if fileutils.IsPathExists(filepath.Join(workingDir, "a.in"), false) {
+		t.Error("File a.in should not have been downloaded")
+	}
+	if fileutils.IsPathExists(filepath.Join(workingDir, "b.in"), false) {
+		t.Error("File b.in should not have been downloaded")
+	}
+	if !fileutils.IsPathExists(filepath.Join(workingDir, "c.tar.gz"), false) {
+		t.Error("Missing file c.tar.gz")
+	}
+	downloadParams.Explode = true
+	// scenario 1: explode in case of checksum download (file exists locally)
```

Comment:
I believe that the scenario 1 and scenario 2 comment are not clear enough for the random reader.
I think we should elaborate more explain what we're trying test here.